code/greedy_algorithms/src/kruskal_minimum_spanning_tree/kruskal.cpp

```diff
@@ -0,0 +1,52 @@
+#include <algorithm>
+#include <vector>
+
+namespace kruskal_impl{
+    using namespace std;
+
+    template<typename _ForwardIter, typename _ValueType>
+    pair<_ValueType, _ValueType> edgeFuncPolicy(_ForwardIter it);
+ 
+    template<typename _ValueType>
+    struct UnionFindPolicy
+    {
+        void merge(_ValueType a, _ValueType b);
+
+        bool connected(_ValueType a, _ValueType b);
+    };
+}
+
+template<typename _HeapedContainer,
+         typename _HeapCompare,
+         typename _EdgeFunc,
+         typename _UnionFind>
+std::vector<typename _HeapedContainer::value_type>
+kruskal(const size_t minimum_edge_count,
+        _HeapedContainer edges,
+        _HeapCompare compare,
+        _EdgeFunc ef,
+        _UnionFind union_sets)
+{
+    using namespace std;
```

Comment:
What advantage does this give you? You use something with std only once/twice, so writing std:: would be clearer in the long run.

Reply:
I think this  implement is not good enough, so I will take care of it next time.